service/matching/taskListManager.go

```diff
@@ -372,13 +378,24 @@ func (c *taskListManagerImpl) GetTaskContext(
 		WorkflowId: common.StringPtr(task.WorkflowID),
 		RunId:      common.StringPtr(task.RunID),
 	}
+
+	// reload from domainCache in case it got empty result during construction
+	if len(c.domainName) == 0 {
```

Comment:
can you create a helper for this whole block of code which checks if domainName is empty and if yes, populates it in a thread safe manner ?

Reply:
I don't get this. Probably need to talk offline.

Reply:
Updated